File: javascript/space-age/space-age.js
export const SpaceAge = class {
  constructor(seconds) {
    this.seconds = seconds;
  }

  onEarth() {
    return parseFloat((this.seconds / (60*60*24*365.25)).toFixed(2));
  }

  onMercury() {
    return parseFloat((this.seconds / (60*60*24*365.25) / 0.2408467).toFixed(2));
  }

  onVenus() {
    return parseFloat((this.seconds / (60*60*24*365.25) / 0.61519726).toFixed(2));
  }

  onMars() {
    return parseFloat((this.seconds / (60*60*24*365.25) / 1.8808158).toFixed(2));
  }

  onJupiter() {
    return parseFloat((this.seconds / (60*60*24*365.25) / 11.862615).toFixed(2));
  }

  onSaturn() {
    return parseFloat((this.seconds / (60*60*24*365.25) / 29.447498).toFixed(2));
  }

  onUranus() {
    return parseFloat((this.seconds / (60*60*24*365.25) / 84.016846).toFixed(2));
  }

  onNeptune() {
    return parseFloat((this.seconds / (60*60*24*365.25) / 164.79132).toFixed(2));
  }
}

File: ruby/space-age/space_age.rb
class SpaceAge
  attr_reader :on_earth, :on_mercury, :on_venus, :on_mars, :on_jupiter, :on_saturn, :on_uranus, :on_neptune

  def initialize(seconds)
    @seconds = seconds
    @on_earth = (seconds / 31557600.00)
    @on_mercury = (@on_earth / 0.2408467)
    @on_venus = (@on_earth / 0.61519726)
    @on_mars = (@on_earth / 1.8808158)
    @on_jupiter = (@on_earth / 11.862615)
    @on_saturn = (@on_earth / 29.447498)
    @on_uranus = (@on_earth / 84.016846)
    @on_neptune = (@on_earth / 164.79132)
  end
end
